code/game/objects/structures/dropship_equipment.dm

```diff
@@ -1039,3 +1042,328 @@
 	deployed_table.layer = ABOVE_OBJ_LAYER + 0.01 //make sure its directly ABOVE the layer
 	deployed_table.loc = loc
 	icon_state = "table2-idle"
+
+//////////////// RAPPEL SYSTEM /////////////////
+
+//Rappel target selection action
+/datum/action/innate/rappel_designate
+	name = "Designate rappel point"
+	action_icon = 'icons/mecha/actions_mecha.dmi'
+	action_icon_state = "mech_zoom_on"
+	var/obj/structure/dropship_equipment/rappel_system/origin
+
+/datum/action/innate/rappel_designate/Activate()
+	var/mob/C = target
+	var/mob/camera/aiEye/remote/remote_eye = C.remote_control
+	var/turf/target_turf = get_turf(remote_eye)
+	var/area/target_area = get_area(target_turf)
+
+	if(origin.retracting) //We're already retracting a rope!
+		return
+
+	if(origin.disabled || origin.cord_damaged || target_turf.density || target_area.ceiling >= CEILING_DEEP_UNDERGROUND)
+		playsound(origin, 'sound/machines/buzz-two.ogg', 25)
+		return
+	if(origin.hooked)
+		playsound(origin, 'sound/machines/buzz-sigh.ogg', 25)
+		return
+
+	C.playsound_local(origin, 'sound/effects/binoctarget.ogg', 35)
+
+	if(origin.unlocked)
+		origin.pre_retract()
+		return
+
+
+	if(!do_after(C, 5 SECONDS, TRUE, remote_eye, BUSY_ICON_BAR))
+		return
+
+	origin.deploy_rope(target_turf)
+	playsound(origin, 'sound/machines/chime.ogg', 50, FALSE)
+
+
+
+//The actual system you put on the tadpole
+/obj/structure/dropship_equipment/rappel_system
+	equip_category = DROPSHIP_CREW_WEAPON
+	name = "rappel deployment system"
+	desc = "A system that deploys rappel ropes to go up or down fast, without the need for the Tadpole to land. You need to designate the rappel point at the navigation computer."
+	dropship_equipment_flags = IS_INTERACTABLE
+	icon_state = "rappel_hatch_locked"
+	point_cost = 300
+	coverage = 0
+	max_integrity = 300
+	anchored = TRUE
+	density = FALSE
+	///The rappeling rope we use
+	var/obj/effect/rappel_rope/tadpole/rope
+	///Whether marines can use the rope or not (i.e, if the rope is deployed)
+	var/unlocked = FALSE
+	///Whether xenos have temporarily disabled the system or not
+	var/disabled = FALSE
+	///Whether the system's cord needs replacing or not
+	var/cord_damaged = FALSE
+	///Smoke particle holder for when it's disabled
+	var/obj/effect/abstract/particle_holder/disabled_smoke
+	///Whether the system is currently retracting a rope or not
+	var/retracting = FALSE
+	///Whether a xeno is currently disabling the system or not
+	var/hooked = FALSE
```

Comment:
Do all these binary vars actually need to exist? They could probs be just flags, or potentially just a general state var.

Reply:
Def should be state var / flags